pkg/oc/cli/set/env.go

```diff
@@ -342,9 +343,9 @@ func (o *EnvOptions) RunEnv() error {
 	errored := []*resource.Info{}
 	for _, info := range infos {
 		name := getObjectName(info)
-		ok, err := o.UpdatePodSpecForObject(info.Object, clientcmd.ConvertInteralPodSpecToExternal(func(spec *kapi.PodSpec) error {
+		ok, err := o.UpdatePodSpecForObject(info.Object, func(spec *corev1.PodSpec) error {
 			resolutionErrorsEncountered := false
-			containers, _ := selectContainers(spec.Containers, o.ContainerSelector)
+			containers, _ := selectContainersV1(spec.Containers, o.ContainerSelector)
```

Comment:
`selectContainers` should work with external versions, `selectContainersInternal` with internals.